ext/mbstring/libmbfl/mbfl/mbfilter.c

```diff
@@ -855,8 +855,19 @@ mbfl_strpos(
 		needle_u8 = needle;
 	}
 
-	if (needle_u8->len < 1) {
-		result = (size_t) -8;
+	if (needle_u8->len == 0) {
+		if (reverse) {
+			result = haystack_u8->len - offset;
+			/* This conditions should be handled in the calling code */
+			if (result < 0) {
```

Comment:
Result is an unsigned number, so this check can't work, please use a similar check as below instead.

Reply:
I wrote something I think is more robust within this branch but I've just realized that I could well live outside and be a general check.

Reply:
I wrote something I think is more robust within this branch but I've just realized that I could well live outside and be a general check.